src/server/data/item/queries.ts

```diff
@@ -11,3 +16,27 @@ export async function getAllItems(): Promise<Item[]> {
     throw error;
   }
 }
+
+export async function getItem(id: number): Promise<Item | undefined> {
+  try {
+    const result = await db.select().from(items).where(eq(items.id, id));
+    return result[0];
+  } catch (error) {
+    console.error("Error fetching item:", error);
+    throw error;
+  }
+}
+
+export async function getItemEmail(id: number): Promise<string | null> {
+  try {
+    const result = await db
+      .select({ email: items.email })
+      .from(items)
+      .where(eq(items.id, id));
+
+    return result[0]?.email;
```

Comment:
```suggestion
const result = await db.query.items.findFirst({
    columns: {
      email: true, // Select only the email column
      },
    where: eq(items.id, id), // Filter by the provided id
});
```
Look into [this](https://orm.drizzle.team/docs/rqb#find-first)

Though your approach also works!!! Either work, I don't mind either just something to look into :3